### DISCUSSION TITLE
connection.dropDb() not works

### DISCUSSION THREAD

**Discussion by Achi-Z** [ujjwalguptaofficial/JsStore#253] [Q&A] [answered]
jsstore is awesome, thank you for it :)

when use it, i got a little problem here :

using your example code 'JsStore_Demo', it success to create and connect,

but fail to drop db by your exmaple code :

connection.dropDb().then(function() {
    console.log('Db deleted successfully');
}).catch(function(error) {
    console.log(error);
});;

error info is ---

{message: "Cannot read properties of undefined (reading 'name')", type: 'TypeError'}
message: "Cannot read properties of undefined (reading 'name')"
type: "TypeError"
[[Prototype]]: Object
constructor: ƒ Object()
hasOwnProperty: ƒ hasOwnProperty()
isPrototypeOf: ƒ isPrototypeOf()
propertyIsEnumerable: ƒ propertyIsEnumerable()
toLocaleString: ƒ toLocaleString()
toString: ƒ toString()
valueOf: ƒ valueOf()
__defineGetter__: ƒ __defineGetter__()
__defineSetter__: ƒ __defineSetter__()
__lookupGetter__: ƒ __lookupGetter__()
__lookupSetter__: ƒ __lookupSetter__()
__proto__: (...)
get __proto__: ƒ __proto__()
set __proto__: ƒ __proto__()
length: 1
name: "set __proto__"
arguments: (...)
caller: (...)
[[Prototype]]: ƒ ()
[[Scopes]]: Scopes[0]

**Comment by ujjwalguptaofficial** [comment]
@Achi-Z could you provide a reproducable repo ? I have added test cases for dropDb and it has passed, so there is a chance something is different in your scenario.

**Comment by Achi-Z** [comment]
`
var connection = new JsStore.Connection(new Worker('jsstore.worker.js'));

var dbName = 'JsStore_Demo';
var tblProduct = {
	name: 'Product',
	columns: {
		// Here "Id" is name of column
		id: {
			primaryKey: true,
			autoIncrement: true
		},
		itemName: {
			notNull: true,
			dataType: "string"
		},
		price: {
			notNull: true,
			dataType: "number"
		},
		quantity: {
			notNull: true,
			dataType: "number"
		}
	}
};
var database = {
	name: dbName,
	tables: [tblProduct]
}

async function initDb() {
	const isDbCreated = await connection.initDb(database);
	if (isDbCreated === true) {
		console.log("db created");
	} else {
		console.log("db opened");
	}
}

// initDb();



async function removeDbA() {
	const isDbCreated = await connection.dropDb();
	if (isDbCreated === true) {
		console.log("db droped A");
	} else {
		console.log("db droped err A");
	}
}
removeDbA();
async function removeDbB() {
	const isDbCreated = await connection.dropDb(database);
	if (isDbCreated === true) {
		console.log("db droped B");
	} else {
		console.log("db droped err B");
	}
}
removeDbB();
async function removeDbC() {
	const isDbCreated = await connection.dropDb(tblProduct);
	if (isDbCreated === true) {
		console.log("db droped C");
	} else {
		console.log("db droped err C");
	}
}
removeDbC();`




all action of droping are not work :(

**Comment by ujjwalguptaofficial** [comment, marked as answer]
that is because you are not initiating the database, you need to initiate db first -> then you can call dropDb. 

If you want to delete three db, initiate each db and then delete.  you can create multiple instance of jsstore for parallely deleting db.
